cobalt/devinfra/kokoro/bin/dind_build.sh

```diff
@@ -42,17 +42,28 @@ fi
 # Load common routines (we only need run_gn and ninja_build).
 . $(dirname "$0")/common.sh
 
-# Using repository root as work directory.
 WORKSPACE_COBALT="${KOKORO_ARTIFACTS_DIR}/git/src"
-cd "${WORKSPACE_COBALT}"
 
-configure_environment
-
-# Build Cobalt.
 pipeline () {
   local out_dir="${WORKSPACE_COBALT}/out/${TARGET_PLATFORM}_${CONFIG}"
-  run_gn "${out_dir}" "${TARGET_PLATFORM}" "${EXTRA_GN_ARGUMENTS:-}"
-  ninja_build "${out_dir}" "${TARGET}"
+  local src_dir="${KOKORO_ARTIFACTS_DIR}/chromium/src"
+  mkdir -p "${src_dir}"
+  cp -aT "${WORKSPACE_COBALT}" "${src_dir}"  # avoid CIPD invalid cross-device link
```

Comment:
why are we copying the checked out code here? 

shouldn't we be able to operate on the checkout at $WORKSPACE_COBALT ?

Reply:
talked with Dana, thee reason for this is nuanced:

1. the directory for our checkout is `/tmpfs/src/git/src` which is its own mount point
2. running the CIPD commands from this directory causes a hardlink to bee created in the `/tmpfs/src/git` directory which isn't allowed since it crosses a filesystem boundary

for now it's easier to just copy the directory

Reply:
Since the operation is rather cryptic I think it would be useful to add this context in a comment.

Reply:
Took another look at these lines. If you resolve all the variables:

`KOKORO_ARTIFACTS_DIR=/tmpfs/src`
`WORKSPACE_COBALT=/tmpfs/src/git/src`
`src_dir=/tmpfs/src/chromium/src`

This operation now first copies `/tmpfs/src/git/src` to `/tmpfs/src/chromium/src` and later the `out` folder in the latter is copied back to the former. Is there a reason you are copying the folders instead of just moving them? Kokoro archives all files under `KOKORO_ARTIFACTS_DIR` that were added after cloning any sources.

I think you will need to either move or delete `/tmpfs/src/chromium` after the build to ensure Kokoro does not try to archive it.

Reply:
TLDR:

1. leave a comment that explains this step more clearly to the reader: "We copy the original gerrit checkout directory from /tmpfs/src/git/src to a new directory /tmpfs/src/chromium/src. This is because /tmpfs/src/git/src is it's own mount point for a device. When CIPD runs, it creates a folder /tmpfs/src/git/.cipd which is on a different device from the original directory. This prevents it from running the steps that create hardlinks. Since we have no way to control how CIPD operates at the time, we move the source to another directory so that when CIPD runs the source directory and the .cipd directory are colocated on the same device"

2. Delete /tmpfs/src/chromium

Reply:
Yep, I figured that out but am reassured to hear the same from you.

Reply:
I put the longer explanation in our internal issue tracker.